### DISCUSSION TITLE
[SOLUTION] To fix the 401 error, install this fixed version of UnDiscord. (version 5.2.3.with.fixes)

### DISCUSSION THREAD

**Discussion by gaveitatry2** [victornpb/undiscord#627] [2. Discussions]
**DISCLAIMER: This fix has nothing to do with 403 errors or index.json files. So if you need help with 403 errors or index.json files, please try elsewhere. Thank you.**
‎
‎
### SOLUTION - INSTALL THE FIXED UNDISCORD:

**Install the new, fixed Undiscord here (version 5.2.3.with.fixes):**

[https://greasyfork.org/en/scripts/502561-undiscord](https://greasyfork.org/en/scripts/502561-undiscord)

‎

### ALTERNATIVE SOLUTION - DELETE DISCORD PASSWORD AND MANUALLY ADD AUTHENTICATION TOKEN

**Step #01 - Remove Discord passwords from your browser (this is what causes Undiscord to break).**

Go to your Passwords manager in your browser.
Search for and delete all passwords for Discord.
Note: Keep in mind to never let your Discord password be saved again in the future, as this will cause Undiscord to break again.

**Step #02 - Get your Authorization Token that your Discord password accidentally replaced.**

Refresh the Discord page.
Press F12 to pull up the Developers Tools window.
Click on the Network Tab.
Click on the XHR tab underneath.
Type"api" (without the quotation marks) in the filter box if you need to narrow down the search results.
Click on one of the search results for Discord.
Note: If you don't see any Discord search results, then refresh the Discord page and follow the other steps again. Keep doing this until you see a Discord search result in the Network > XHR section in the Developer Tools window.
Click on the Headers tab.
Scroll down until you see Authorization: (some long string of letters and numbers).
Right-click that and selected Copy Value.
Note: It is recommended to open up a Notepad document and paste the value that you just copied to make sure that the Authorization Token was copied correctly.
Note: You need to manually get your Authorization Token using this above method, because pressing the Authorization Token Fill button may just give you an error message and won't work.

**Step #03 - Paste your Authorization Token in your Undiscord window.**

Open up the Undiscord window and paste the Authorization Token that you got from the above steps.

**Comment by GauravASB** [comment]
okay this works. Thanks, Bud.

**Comment by Pestermess** [comment]
This isn’t working for me, even though I’m using Mozilla Firefox!

**Reply by njargh** [reply]
Hey, i just made a new thread with an easier way to fix that issue. :)

**Reply by Pestermess** [reply]
The main issue with all these replies is that the developer isn’t addressing the problem personally. Why should we have to jump through hoops when the extension worked perfectly fine for years? I don’t have the time to troubleshoot why it suddenly stopped working. It should either work out of the box or be put on hold until it’s fixed. This situation is causing a major headache for me and my mods, as nothing seems to be functioning as promised.

**Reply by jasonseeb** [reply]
@PesterMess These people work for free on their own time. Why don't you work on something for free and then have other come in to tell you to fix your free program? I'm sure you would love it if that happened.

**Reply by jasonseeb** [reply]
"But I also can relate to jasonseeb because I get tired of developers who publish things as "working"" I believe this is a typo. PesterMess is the one who gets tired of developers who publish things as "working".

I understand that if I do something for free, then there are no expectations. If something takes 5-10 seconds to fix, then the person complaining about it should take the time to learn to program. The knowledge needed should not take long since it is "an assumed" 5-10 seconds to fix. I will say it takes a couple of hours to learn how to fix it. The people who program these things took years to learn what they do.

If I was given a free product that did not work for me, I would not complain about it; I would just go on living knowing that I spent nothing on it.

**Reply by Pestermess** [reply]
Unfortunately, I'm not in a position to drop everything and learn how to fix this issue on my own. My team and I manage multiple Discord servers, and on top of that, we all have full-time jobs. Despite our best efforts, we haven't been able to resolve the error following the tutorial provided.

Moreover, we’re not willing to resort to downloading a second browser just to work around a faulty extension, especially when the developer has provided minimal support and communication options. It’s quite disappointing that the developer has left so little room for support or dialogue.

At this point, we will have to escalate the issue by reporting the GitHub repository until the developer addresses and resolves these problems.

**Reply by jasonseeb** [reply]
"My team and I manage multiple Discord servers, and on top of that, we all have full-time jobs."
Does the programmer have a full-time job the same as you guys and doesn't have time to fix this? This doesn't pay him or her anything.

"Despite our best efforts, we haven't been able to resolve the error following the tutorial provided."
I have successfully tested this on Firefox and Chrome. They both work. Can I help you? You might be missing something from the instructions. I would even offer help over Discord to screen share the issue and how you are trying to solve it.

"At this point, we will have to escalate the issue by reporting the GitHub repository until the developer addresses and resolves these problems."
The repository is not required to work. What are you going to report? There are countless repositories in the same state. None of them can be reported because there is no guarantee that anything will work, nor is it required by GitHub that you maintain a project or meet any deadlines to make anything work.

**Reply by ghost** [reply]
@jasonseeb I tried everything as i can do it is still not working u can help me if u want

**Comment by gaveitatry2** [comment]


**Reply by Pestermess** [reply]
Add me on discord and ill show you what's happening: **pestermess_87491** Because my password is NOT SAVED on the browser and I click fill and nothing works regardless!

**Reply by jasonseeb** [reply]
@gaveitatry2 I'd like to thank you for taking the time to make a video.

**Comment by gaveitatry2** [comment]


**Reply by jasonseeb** [reply]
I completely agree the fix is probably fairly easy. The thing is I thanked you yesterday because you took the time to do the video. I didn't need the video as your initial instructions was good enough for me. The person who was complaining the most has either ignored your video or has seen it and not thanked you. That's why I was thinking about taking the time to either make a fork and fix it or fix this branch. But after seeing the treatment and attacks against people doing free work, it really dampens the desire.

**Comment by sn-o-w** [comment]
There's a way without deleting your browser passwords!

1) Search the following lines and locally modify them:

> ~~`**#undiscord input[type="password"],**`~~
> ~~**=>**~~
> ~~`**#undiscord input[type="text"],**`~~

**_EDIT: Actually `#undiscord input[type="text"],` is not needed, since is already mentioned in the script.
So you can just remove the line `#undiscord input[type="password"],`, don't bother to modify it in `#undiscord input[type="text"],`_**

> `**<input class="input" id="token" type="password" autocomplete="dont" priv>**`
> **=>**
> `**<input class="input" id="token" type="text" priv>**`

**_EDIT: ^ This change is still needed._**

And it should work!

- **[OPTIONAL STEP]** You can also make these changes to prevent the user script from stopping after a while: https://github.com/victornpb/undiscord/pull/619/files

I highly recommend doing the optional step as well.

**Reply by ghost** [reply]
@gaveitatry2 not working for me can u help?

**Reply by ghost** [reply]
@gaveitatry2 why you seem pissed buddy? i just wanted help from you I'm not lying or trolling you I tried that different version u sended me rn and it still errors at some point idk how to send a vid here I just created an ACC cuz nobody I know was be able to help me about this situation I can share my screen in discord if u want to help realmfg is my discord

**Reply by ghost** [reply]
@gaveitatry2 you using discord? i can share my screen in discord and u can see everything that error it would be much easier but its all up to you so

**Reply by ghost** [reply]
@gaveitatry2 dude dont worry im not fooling you i mean for me giving discord accounts is not that a big deal but if it is for you then fine ill ask snow for help if he don't ill make the video for you

**Reply by ghost** [reply]
@gaveitatry2 https://github.com/victornpb/undiscord/pull/619 this and the one that yours gave me the Error searching messages, API responded with status 403!
	{"message":"Erişim Bulunmuyor","code":50001}
CoreException	{}

https://github.com/victornpb/undiscord/issues/655 and this one gave me CoreException	{"stack":"TypeError: Cannot create property 'desc' on number '0'\n    at UndiscordCore.run (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=Undiscord.user.js&id=736fcb82-5082-47fd-bd29-7fd5e8349251:773:50)\n    at async UndiscordCore.runBatch (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=Undiscord.user.js&id=736fcb82-5082-47fd-bd29-7fd5e8349251:690:8)\n    at async HTMLButtonElement.startAction (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=Undiscord.user.js&id=736fcb82-5082-47fd-bd29-7fd5e8349251:1856:8)","message":"Cannot create property 'desc' on number '0'"}

I manually put my token and my browser don't have saved passwords I also tried on firefox and still didn't work
(P.S I'm trying to delete all my messages in discord so the index.json file)

**Reply by ghost** [reply]
@gaveitatry2 its only ''realmfg''

**Comment by ghost** [comment]
@sn-o-w can you help ? my discord is realmfg I can share screen and it make things easier ig

**Comment by gaveitatry2** [comment]
Unsubscribed because of idiots. PS - I switched to Discrub. It's an extension on Chrome that is actively being maintained, has high ratings, and is also open source. I used it to delete 40,000 DMs. I toggle the extension off when I'm not using it, just to be safe.
